File: PH2007SDK/developpers/VBTut2/Collector.vb
'Tutorial done by
' 
'Richard Clark
'Project Hoshimi Lead Manager
'Contact at ori@c2i.fr
' 
Imports PH.Common

<Characteristics(ContainerCapacity:=20, _
    CollectTransfertSpeed:=5, _
    Scan:=5, _
    MaxDamage:=0, _
    DefenseDistance:=0, _
    Constitution:=20)> _
  Public Class Collector
    Inherits PH.Common.NanoCollector
    Implements IActionable

    Public Enum WhatToDoNextAction
        MoveToAZN = 0
        CollectAZN = 1
        MoveToHoshimi = 2
        TransfertToNeedle = 3
    End Enum

    Private m_WhatToDoNext As WhatToDoNextAction = WhatToDoNextAction.CollectAZN
    Public Property WhatToDoNext() As WhatToDoNextAction
        Get
            Return Me.m_WhatToDoNext
        End Get
        Set(ByVal value As WhatToDoNextAction)
            m_WhatToDoNext = value
        End Set
    End Property

    Public Sub DoActions() Implements IActionable.DoActions
        Select Case Me.WhatToDoNext
            Case WhatToDoNextAction.CollectAZN
                Me.CollectFrom(Me.Location, 4)
                Me.WhatToDoNext = WhatToDoNextAction.MoveToHoshimi
            Case WhatToDoNextAction.MoveToHoshimi
                Me.MoveTo(Utils.getNearestPoint(Me.Location, CType(Me.PlayerOwner, myPlayer).HoshimiEntities))
                Me.WhatToDoNext = WhatToDoNextAction.TransfertToNeedle
            Case WhatToDoNextAction.TransfertToNeedle
                Me.TransferTo(Me.Location, 4)
                Me.WhatToDoNext = WhatToDoNextAction.MoveToAZN
            Case WhatToDoNextAction.MoveToAZN
                Me.MoveTo(Utils.getNearestPoint(Me.Location, CType(Me.PlayerOwner, myPlayer).AZNEntities))
                Me.WhatToDoNext = WhatToDoNextAction.CollectAZN
        End Select
    End Sub
End Class

<Characteristics(ContainerCapacity:=100, _
CollectTransfertSpeed:=0, _
Scan:=10, _
MaxDamage:=5, _
DefenseDistance:=10, _
Constitution:=25)> _
Public Class Needle
    Inherits PH.Common.NanoNeedle

End Class

File: PH2007SDK/developpers/CSTut2/Collector.cs
/* Tutorial done by
 * 
 * Richard Clark
 * Project Hoshimi Lead Manager
 * Contact at ori@c2i.fr
 * 
*/
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using PH.Common;
using PH.Map;

namespace CSTut2
{

    [Characteristics(ContainerCapacity = 20,
     CollectTransfertSpeed = 5,
     Scan = 5,
     MaxDamage = 0,
     DefenseDistance = 0,
     Constitution = 20)]
    public class Collector : PH.Common.NanoCollector, IActionable
    {
        public enum WhatToDoNextAction
        {
            MoveToAZN = 0,
            CollectAZN = 1,
            MoveToHoshimi = 2,
            TransfertToNeedle = 3,
        }
        #region IAction Members

        private WhatToDoNextAction m_WhatToDoNext = WhatToDoNextAction.CollectAZN;
        public WhatToDoNextAction WhatToDoNext
        {
            get { return m_WhatToDoNext; }
            set { m_WhatToDoNext = value; }
        }  
        public void DoActions()
        {
            switch (this.WhatToDoNext)
            {
                case WhatToDoNextAction.CollectAZN:
                    this.CollectFrom(this.Location, 4);
                    this.WhatToDoNext = WhatToDoNextAction.MoveToHoshimi;
                    break;
                case WhatToDoNextAction.MoveToHoshimi:
                    this.MoveTo(Utils.getNearestPoint(this.Location, ((myPlayer)this.PlayerOwner).HoshimiEntities));
                    this.WhatToDoNext = WhatToDoNextAction.TransfertToNeedle;
                    break;
                case WhatToDoNextAction.TransfertToNeedle:
                    this.TransferTo(this.Location, 4);
                    this.WhatToDoNext = WhatToDoNextAction.MoveToAZN;
                    break;
                case WhatToDoNextAction.MoveToAZN:
                    this.MoveTo(Utils.getNearestPoint(this.Location, ((myPlayer)this.PlayerOwner).AznEntities));
                    this.WhatToDoNext = WhatToDoNextAction.CollectAZN;
                    break;
            }
        }

        #endregion
    }

    [Characteristics(ContainerCapacity = 100,
    CollectTransfertSpeed = 0,
    Scan = 10,
    MaxDamage = 5,
    DefenseDistance = 10,
    Constitution = 25)]
    public class Needle : PH.Common.NanoNeedle, IActionable
    {
        #region IAction Members

        public void DoActions()
        {}
        #endregion
    }

}
